algorithms/Python/dynamic_programming/catalan_sequence.py

```diff
@@ -0,0 +1,37 @@
+# A dynamic programming based function to find nth
+# Catalan number
+
+"""
+Output: 
+The first 10 terms of Catalan sequence are :
+1 1 2 5 14 42 132 429 1430 4862
+"""
+
+
+def catalan(n):
```

Comment:
Add doctests like the one in interval_scheduling.py, binary_search.py etc

Reply:
I have added the doctests. Please check.